### DISCUSSION TITLE
Add rack for Depth Charge Shell

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#7461] [Ideas and suggestions]
Depth charges rolling on the your floor?
They’ll think we’re crazy.
_P.S. Dear developers! Please resolve this!_

**Update:**
v0.18.11.0
- Depth charges can be stored in coilgun ammo shelves.

**Comment by ubertpendragon** [comment]
Note that crate shelves are a perfect fit; depth charges only need the "crate" tag

![image](https://user-images.githubusercontent.com/73229309/143083134-e50d7011-0d08-4702-be96-ffe1ec597791.png)

**Comment by MaaiG** [comment]
> crate shelves

Yea! Thanks! This will solve many problems with existing design and lack of free space. Perhaps this is the ideal solution!

\Content\Items\Weapons\depthcharge.xml
```
Line 21:   <Item name="" identifier="depthchargeshell" tags="depthchargeammo" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
line 56:   <Item name="" identifier="nucleardepthcharge" tags="depthchargeammo" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
line 111:   <Item name="" identifier="depthdecoyshell" tags="depthchargeammo,decoy" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
line 151:   <Item name="" identifier="nucleardepthdecoy" tags="depthchargeammo,decoy" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
```
to
```
Line 21:   <Item name="" identifier="depthchargeshell" tags="depthchargeammo,crate" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
Line 56:   <Item name="" identifier="nucleardepthcharge" tags="depthchargeammo,crate" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
Line 111:   <Item name="" identifier="depthdecoyshell" tags="depthchargeammo,decoy,crate" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
Line 151:   <Item name="" identifier="nucleardepthdecoy" tags="depthchargeammo,decoy,crate" category="Misc" sonarsize="5" scale="0.5" impactsoundtag="impact_metal_heavy">
```

**Reply by JeroonBosch** [reply]
This kind of approach can give trouble in the future for us (as well as modders) because depth charges are not actually crates, so it's safer to just tag depth charges as 'cratesized'  and allow shelves than contain crates also able to contain crate-sized items (which is the approach I went with)

**Comment by gerakis** [comment]
I just came here to suggest this. Also an autoloader for charges would be good, like how the railgun can load up 5 shots. We've been doing this since WW2 so it seems like space age sailors would have it.


![Depth_Charge__rack](https://user-images.githubusercontent.com/16128947/144659613-8f1a19cd-cb3b-40fb-a1b5-cdc05b51c34f.jpg)
![USS_Cassin_Young_depth_charges](https://user-images.githubusercontent.com/16128947/144659625-7787b14d-adc0-4162-a341-414681bd81e8.jpg)

**Reply by MaaiG** [reply]
Yeah...
I'm doing this (same for depth charges.) - The ammunition [hoists system](https://github.com/Regalis11/Barotrauma/discussions/7432):

https://user-images.githubusercontent.com/93348499/144679745-29ed575f-145a-4f30-b87f-d032eff5d818.mp4

> So it seems like space age sailors would have it.

Maybe, it's only a matter of time. But just how long to wait!

**Comment by MaaiG** [comment]
Captain's log, star date 2021.
...
_Bomb racks are still not freely available.
Yesterday, my junior mechanic tripped over a depth-charge bomb and hit his head on the detonator of another bomb.
Good thing he wasn’t wearing his hard hat...
Ordered the engineer for take off all detonating fuses.
What’s going on on that goddamn moon?_
...

**Comment by MaaiG** [comment]
`Beacon:` **Hope**
`Recording:` **01001110 01101111**
`Type:` abyss signal
`Class:` reverb
`Source:` unknown
`Transcription:` yes

_Captain’s log, star date 2022J171523.
Voice record.
...
It was inevitably. Why the hell did I go into that part of the sub. I should have stayed in cabin. Near the ammunition depots it would be over instantly...
Now I’m lying damp with a bulkhead around my neck. It trembles and seems to be moving in the dark and I can’t move. Metal junk squeeze on all sides more and more. I knew this would happen. I knew! It had to hap..._
...
`Record damaged. Recovery:`
...
_...hope we’re falling into the abyss and not stuck on... 
...some lump. Let it end.
Dam...
...you..._
...
`Record damaged. Irreversibly.`
`Signal fading.`

**Comment by MaaiG** [comment]
_Captain’s log, star date 2022J
...
`Test` 0.17.16.0_

https://user-images.githubusercontent.com/93348499/173186176-a3760884-9279-4418-bea4-132240e3d80c.mp4

`Conclusion`
_Don't lift the bombs from the floor! It can cause them to fall! Let them roll on the floor..._

**Comment by MaaiG** [comment]
Captain’s log, star date 2022J 0.18.11.0
- _Depth charges can be stored in coilgun ammo shelves._

**Comment by Ivanzyfer** [comment]
Bombs are too useless in my opinion. It's difficult for them to hit someone only if they stand still and wait. Why make them at all?
